packages/desktop/components/popup/popups/EvmTransactionFromDappPopup.svelte

```diff
@@ -28,17 +28,23 @@
     import { BASE_TOKEN_ID } from '@core/token/constants'
     import { checkActiveProfileAuthAsync } from '@core/profile/actions'
     import { LedgerAppName } from '@core/ledger'
-    import { DappVerification } from '@auxiliary/wallet-connect/enums'
+    import { DappVerification, RpcMethod } from '@auxiliary/wallet-connect/enums'
 
     export let preparedTransaction: EvmTransactionData
+    export let rawTransaction: string
     export let chain: IChain
     export let dapp: IConnectedDapp
-    export let signAndSend: boolean
     export let verifiedState: DappVerification
+    export let method: RpcMethod.EthSendTransaction | RpcMethod.EthSignTransaction | RpcMethod.EthSendRawTransaction
     export let callback: (params: CallbackParameters) => void
 
     const { id } = chain.getConfiguration()
-    $: localeKey = signAndSend ? (isSmartContractCall ? 'smartContractCall' : 'sendTransaction') : 'signTransaction'
+    $: localeKey =
+        method === RpcMethod.EthSignTransaction
+            ? 'signTransaction'
+            : isSmartContractCall
+              ? 'smartContractCall'
+              : 'sendTransaction'
```

Comment:
Use dictionary convention

Reply:
The dictionary would be unnecessarily complex because we would have the method AND this `isSmartContractCall` boolean as layered keys.